domain/policy.go

```diff
@@ -311,6 +312,21 @@ type PolicyAppealConfig struct {
 	AllowCreatorDetailsFailure bool `json:"allow_creator_details_failure" yaml:"allow_creator_details_failure"`
 }
 
+type AppealMetadataSource struct {
+	Name        string                `json:"name" yaml:"name"`
+	Description string                `json:"description,omitempty" yaml:"description,omitempty"`
+	Type        string                `json:"type" yaml:"type"`
+	Config      *AppealMetadataConfig `json:"config,omitempty" yaml:"config,omitempty"`
```

Comment:
`Config` should have `interface{}` type because it can be in any struct form depending on the `Type`.

In the policy creation, we can stringify the entire `Config` value and then encrypt before db insertion

In the appeal creation when we use `Config` for an `http` metadata, then we will assert the Config to be `http.HttpClientConfig`